client/src/components/WinLoss.svelte

```diff
@@ -0,0 +1,201 @@
+<script lang="ts">
+  import { authStore as user } from "$stores/auth";
+  import type { PrivatePlayer, Lobby } from "../../../functions/src/firestore-types/lobby";
+  import { onMount } from "svelte";
+  import { lobbyReturn } from "$lib/firebase/firestore-functions";
+  import { getPrivatePlayerCollection } from "$lib/firebase/firestore-collections";
+  import { doc, getDoc } from "firebase/firestore";
+  import type { DocumentReference } from "firebase/firestore";
+
+  export let lobbyDocRef: DocumentReference<Lobby>;
+  export let lobbyCode: string;
+  export let lobby: Lobby;
+
+  let end: "Cat Win" | "Cat Lose" | "Catfish Lose" | "Catfish Win" | null = null;
+  let catfishList: string = "";
+  let catfishes: string[];
+  onMount(async () => {
+    // format a string list to have commas and "and" when needed
+    const formatter = new Intl.ListFormat("en", { style: "long", type: "conjunction" });
+    // get the list of catfishes from lobby.ts and put them into a variable
+    catfishes = lobby.players
+      .filter((p) => {
+        return p.role == "CATFISH";
+      })
+      .map((c) => c.displayName);
+    catfishList = formatter.format(catfishes);
+
+    // grab the current user's role
+    const myRole = (await getDoc(doc(getPrivatePlayerCollection(lobbyDocRef), $user?.uid))).data() as PrivatePlayer;
+    // update end variable depending on the winner type and the current player's role
+    if (myRole.role == "CAT") {
+      if (lobby.winner == "CAT") {
+        end = "Cat Win";
+      } else if (lobby.winner == "CATFISH") {
+        end = "Cat Lose";
+      }
+    } else {
+      if (lobby.winner == "CAT") {
+        end = "Catfish Lose";
+      } else if (lobby.winner == "CATFISH") {
+        end = "Catfish Win";
+      }
+    }
+  });
+</script>
+
+<div class="container">
+  <!-- Check the user's role and display the correct win or loss page accordingly -->
+  {#if end == "Cat Win"}
+    <div class="banner">
+      <h2>Hooray!</h2>
+      <h2>You have sniffed out the cat fish!</h2>
+    </div>
+
+    <div class="image">
+      <img src="https://www.azpetvet.com/wp-content/uploads/Happy-cat-1024x711.jpg" alt="clown fish" />
+    </div>
+
+    <div class="displayname">
+      {#if catfishes.length > 1}
+        <h2>{catfishList} were sentenced to 9 lives in purrison!</h2>
+      {:else}
+        <h2>{catfishList} was sentenced to 9 lives in purrison!</h2>
+      {/if}
+      <!-- {#if isHost} -->
+      <button
+        on:click={() => {
+          lobbyReturn({ code: lobbyCode });
+        }}>Return to Lobby</button
+      >
```

Comment:
in `winloss.ts` you say:
```ts
if (context.auth.uid !== uids[0]) {
  return { error: "Not allowed to return to lobby!" };
}
```
if that's how you are going to make it work then this button should only be visable to the host

Reply:
I forgot to comment it out, but since were adding timers now, the lobby return button is not going to be used. I just have the lobbyReturn function to test until the timers get added

Reply:
Gotcha.